saml/models/core/request.go

```diff
@@ -0,0 +1,169 @@
+package core
+
+import (
+	"encoding/xml"
+	"strings"
+	"time"
+)
+
+// See 3.2.1 http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf
+type StatusRequestType struct {
+	RequestResponseCommon
+}
+
+// See 3.4.1 http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf
+// TODO Finish this
+type AuthnRequest struct {
+	XMLName xml.Name `xml:"urn:oasis:names:tc:SAML:2.0:protocol AuthnRequest"`
+
+	StatusRequestType
+
+	Subject              *Subject
+	NameIDPolicy         *NameIDPolicy `xml:",omitempty"`
+	Conditions           *Conditions
+	RequestedAuthContext *RequestedAuthnContext
+	Scoping              *Scoping
+
+	ForceAuthn bool `xml:",attr,omitempty"`
+	IsPassive  bool `xml:",attr,omitempty"`
+
+	AssertionConsumerServiceIndex string `xml:",attr,omitempty"`
+	AssertionConsumerServiceURL   string `xml:",attr"`
+
+	// A URI reference that identifies a SAML protocol binding to be used when
+	// returning the Response message.
+	ProtocolBinding ServiceBinding `xml:",attr"`
+
+	AttributeConsumingServiceIndex string `xml:",attr,omitempty"`
+	ProviderName                   string `xml:",attr,omitempty"`
+}
+
+// Subject specifies the requested subject of the resulting assertion(s).
+// If entirely omitted or if no identifier is included, the presenter of
+// the message is presumed to be the requested subject.
+//
+// See 2.4 http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf
+type Subject struct {
+	XMLName xml.Name `xml:"urn:oasis:names:tc:SAML:2.0:assertion Subject"`
+
+	BaseID              *BaseID
+	NameID              *NameID
+	EncryptionID        string
```

Comment:
Wondering if this should be `EncryptedID *EncryptedID` similar to `SubjectConfirmation`? Was looking at [2.4.1 Element <Subject>](http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf).

Reply:
Oh the spelling is wrong, too. 

Yes I think that's a good idea.

Reply:
Pushed a small fix for the type/typo in https://github.com/hashicorp/cap/pull/99/commits/9346964d19a35b81f8cf06016107108591d89ca8.